cmd/swarm/main.go

```diff
@@ -327,26 +331,43 @@ func init() {
 					Action:             resourceCreate,
 					CustomHelpTemplate: helpTemplate,
 					Name:               "create",
-					Usage:              "creates a new Mutable Resource",
-					ArgsUsage:          "<frequency>",
-					Description:        "creates a new Mutable Resource",
-					Flags:              []cli.Flag{SwarmResourceNameFlag, SwarmResourceDataOnCreateFlag, SwarmResourceTopicFlag},
+					Usage:              "creates and publishes a new Mutable Resource manifest",
+					Description: `creates and publishes a new Mutable Resource manifest pointing to a specified user's updates about a particular topic.
+					The topic can be specified directly with the --topic flag as an hex string
+					If no topic is specified, the default topic (zero) will be used
+					The --name flag can be used to specify subtopics with a specific name
```

Comment:
This is a bit of calculated ambiguity. We have two very useful use cases here:
a) when topic + name are used together
* `topic` is supposed to be an arbitrary binary string
* `name`is supposed to be something human-readable and easy to remember, represents a subtopic name
b) when only name is used
* in this case, `name` is like the resource name we had before. This allows to find per-user information: For example, the contact e-mail of a particular ethereum address could be looked up using `name=email` and no topic.

Internally, on the wire, there is only "topic". This CLI interface is a way to help the user build useful topics so others can easily arrive to the same "topic" and get the content.

Reply:
Ok. Where is this documented?

Reply:
The CLI help, we'll put this in the guide too.

Reply:
You mean

> The --name flag can be used to specify subtopics with a specific name.

I'm not sure that really clears it up?

Reply:
I reworded this as follows:
```
Description: `creates and publishes a new Mutable Resource manifest pointing to a specified user's updates about a particular topic.
	The resource topic can be built in the following ways:
	* use --topic to set the topic to an arbitrary binary hex string.
	* use --name to set the topic to a human-readable name.
		For example --name could be set to "profile-picture", meaning this Mutable Resource allows to get this user's current profile picture.
	* use both --topic and --name to create named subtopics.
		For example, --topic could be set to an Ethereum contract address and --name could be set to "comments", meaning
		the Mutable Resource tracks a discussion about that contract.
	The --user flag allows to have this manifest refer to a user other than yourself. If not specified,
	it will then default to your local account (--bzzaccount)`,
```

Reply:
The description is much better.

But.

I can't shake the feeling that this implementation is a but clunky, though. I would imagine having one parameter and using a path-like value would be more intuitive? And maybe in the future we could imagine having hierarchical "names" with possibility for recursive retrieval?

Could the topic be <topic>/<name>/<subname> then thus parsed by the cli? The topic could be 0xsomething to be hex-interpreted, or else parsed as ascii?

Reply:
`--name` takes a string, up to 32 chars.
`--topic` takes hex-encoded bytes, up to 32 bytes (64 hex digits), `0x` prefixed.
Remember you can use one or the other or both.

Internally, the "real" final topic used on the wire will be `--topic XOR --name`, so if you don't use one of the parameters, you are XORing with 0 which leaves the other parameter untouched.

This is the way I have been enjoying MRUs as of late:
```
$ swarm resource update --name "profile-picture" 0x1b2025cd94dbd22777479967a9c3013fb54cf841d4bb7b1640b172e6ed6573ed5d18
```
This is very much readable, and it is clear that would update my profile picture.

To retrieve somebody's profile picture given their address:
`curl http://localhost:8500/bzz-resource:/?user=<0xUSERADDRESS>&name=profile-picture`
Also pretty intuitive.

> I can't shake the feeling that this implementation is a but clunky, though.

Why? You can do just about everything with it.

> I would imagine having one parameter and using a path-like value would be more intuitive?

How you generate the topic is up to you. For example, you could totally ignore `--topic` and use `--name` and put a path there. (path limited to 32 chars)

Alternatively, you can hash your path to a topic for arbitrary-length paths. For example:
```
$ swarm resource update --topic "0x$(echo -n path/to/file/that/is/very/long | sha256sum | cut -c 1-64)" 0x1b20<multihash...>
```
> And maybe in the future we could imagine having hierarchical "names" with possibility for recursive retrieval?

Interesiting. Can you please elaborate in a new issue to take the discussion there? but upfront I don't see why this could not be built as a higher level abstraction or as part of the application. We are discussing a simple CLI here to give entry-level access to MRUs.

> Could the topic be // then thus parsed by the cli?

Sorry, what do you mean by topic = `//` ?

> The topic could be 0xsomething to be hex-interpreted, or else parsed as ascii?

Then you would not have the opportunity to use both simultaneously. Viktor envisioned that given a hash of something (a contract, a content hash, whatever), there would be human-readable subtopics about it, so if I want to give my opinion about a smart contract on 0xabcd123945abcd123945abcd123945abcd123945, I could use the "discussion" subtopic:
```
$ swarm resource update --topic 0xabcd123945abcd123945abcd123945abcd123945 --name "discussion" 0x<my comment>
```
Actual topic on the wire for the above command would be `0xabcd123945abcd123945abcd123945abcd123945 ^ []byte("discussion") = 0xcfa4615a30d8be7b562babcd123945abcd123945000000000000000000000000`
(the zero padding appears because a contract address is 20 bytes long only)